external-import/sentinel-incidents/src/sentinel_incidents_connector/config_variables.py

```diff
@@ -16,6 +16,25 @@ def __init__(self):
         self.load = self._load_config()
         self._initialize_configurations()
 
+    @staticmethod
+    def prepare_iso_format(date: str) -> str | datetime:
```

Comment:
The `prepare_iso_format` function is designed to ensure that the date used in API calls is correctly formatted as a string in ISO 8601 format (for the first run). In this case, we need the date as a string (`string`), not as a `datetime` object (I'll change that), as the API requires this format to accept date filters in queries.

The config must accept two date formats as input:

1. `‘2020-01-01T00:00:00Z’` (full ISO 8601 format)
2. `‘2020-01-01’` (abbreviated format, only the date without the time)

- If the input format is full ISO 8601** (`‘2020-01-01T00:00:00Z’`), the function will return this string directly.
- If the input format is abbreviated** (`‘2020-01-01’`), then the function will enrich it to include the time at midnight (00:00:00) and the suffix `Z` to indicate UTC time, using `.strftime(‘%Y-%m-%dT%H:%M:%SZ’)`. This complete format (`‘2020-01-01T00:00:00Z’`) is required to ensure that the API interprets the date correctly in all cases.
- If the date is not supplied by the user**, then the default date is ‘2020-01-01T00:00:00Z’.

And since you're fond of links, here's another one xD: 
https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior

But if you prefer, I'll change ‘strptime()’ to ‘fromisoformat()’:
```
prepared_date = datetime.strptime(date, "%Y-%m-%dT%H:%M:%SZ").strftime("%Y-%m-%dT%H:%M:%SZ")
prepared_date = datetime.fromisoformat(date).strftime("%Y-%m-%dT%H:%M:%SZ")
```

Reply:
Thanks for the explanations (and the link ^^)

For the sake of readability I would tend not to use the format codes (`%Y-%m-%dT%H:%M:%SZ`) and so prefer something like that, which I find more self-explanatory:
```
>>> from datetime import datetime, timezone
>>> iso_string = '1970-01-01'
>>> iso_datetime = datetime.fromisoformat(iso_string).replace(tzinfo=timezone.utc)
>>> utc_iso_string = iso_datetime.isoformat()
>>> print(utc_iso_string)
1970-01-01T00:00:00+00:00
```

It's just to share ideas, your code is 100% valid too, you don't need to change it 👍

Reply:
Ok I understand, once again, initially I wasn't really looking to refacto everything, but I've taken your opinions into account, it'll be changed ;)